api/src/routes/edit.ts

```diff
@@ -197,8 +197,13 @@ function uploadFile(req, res, next) {
             return;
         }
         try {
+            let fileEntry = files?.file ?? files?.[Object.keys(files || {})[0]];
+            if (Array.isArray(fileEntry)) {
+                fileEntry = fileEntry[0];
+            }
+            const filepath = fileEntry.filepath ?? fileEntry.path ?? fileEntry.filePath;
+            const mimetype = fileEntry.mimetype ?? fileEntry.type ?? "";
```

Comment:
Is this logic attempting to be back-compatible with the older version of Formidable? If so, I don't think that's necessary -- since we're locking in the newer version, I think it would be safe to assume that's what we're going to use. If that's the case, could we simplify to something like:


```suggestion
            let fileEntry = files[0];
            const filepath = fileEntry.filepath;
            const mimetype = fileEntry.mimetype;
```

(That's just a guess -- I am probably getting some structural details wrong... but I think there has to be an easier way than what we have here, since it seems to be accounting for many theoretical possibilities that may not really exist. I'd just add `console.log(files)` to the code to see what the structure looks like, and then adjust the logic to accommodate that).

Reply:
That's pretty much what I had before, but it tripped up the test. I'll switch it back and look into correcting the test instead.

Reply:
Yeah, the test is probably sending in fake data in the old format, so if you change the test's simulated data to match the new reality, then everything will be happy.